library/ziti_ctrl.c

```diff
@@ -169,6 +170,13 @@ static void ctrl_resp_cb(um_http_resp_t *r, void *data) {
         }
 
         resp->new_address = find_header(r, "ziti-ctrl-address");
+
+        const char *instance_id = find_header(r, "ziti-instance-id");
+
+        if (instance_id) {
+            FREE(resp->ctrl->instance_id);
+            resp->ctrl->instance_id = strdup(instance_id);
+        }
```

Comment:
This section is the only part I actually changed. The rest is auto-formatting from CLion (default installation).

Reply:
Removed formatting